Loop/Managers/DeviceDataManager.swift

```diff
@@ -1376,81 +1376,56 @@ extension DeviceDataManager {
             log.error("Remote Notification: Remote Commands not enabled.")
             return
         }
-
-        if let expirationStr = notification["expiration"] as? String {
-            let formatter = ISO8601DateFormatter()
-            formatter.formatOptions =  [.withInternetDateTime, .withFractionalSeconds]
-            if let expiration = formatter.date(from: expirationStr) {
-                let nowDate = Date()
-                guard nowDate < expiration else {
-                    let expiredInterval = nowDate.timeIntervalSince(expiration)
-                    await NotificationManager.sendRemoteCommandExpiredNotification(timeExpired: expiredInterval)
-                    log.error("Remote Notification: Expired: %{public}@", String(describing: notification))
-                    return
-                }
-            } else {
-                log.error("Remote Notification: Invalid expiration: %{public}@", expirationStr)
-                return
-            }
-        }
-        
-        let action: Action
         
+        let command: RemoteCommand
         do {
-            action = try RemoteCommand.createRemoteAction(notification: notification).get()
+            command = try await remoteDataServicesManager.commandFromPushNotification(notification)
         } catch {
             log.error("Remote Notification: Parse Error: %{public}@", String(describing: error))
             return
         }
         
-        log.default("Remote Notification: Handling action %{public}@", String(describing: action))
+        await handleRemoteCommand(command)
+    }
+    
+    func handleRemoteCommand(_ command: RemoteCommand) async {
+        
+        log.default("Remote Notification: Handling command %{public}@", String(describing: command))
         
```

Comment:
The RemoteCommand can communicate with its remote service. This will be helpful in the future where command status will be reported back to the service. 

I'm leveraging that capability here where validate() is now called directly on the command without needing to look up the originating source in this code, as was previously done in validatePushNotificationSource.

Reply:
The only reason I validate the command in each switch case below, rather than in just one place above: I need to use the appropriate error handler which vary for remote carbs vs remote bolus. The overall error handling of commands below is something to consider refactoring in the future but I wanted to keep this PR scoped.